--- entrance-0.9.0.009/src/client/entrance_session.c.orig	2007-04-03 13:36:51.000000000 +0200
+++ entrance-0.9.0.009/src/client/entrance_session.c	2007-09-09 18:38:09.567076559 +0200
@@ -854,7 +854,7 @@
    if (shell && (strlen(shell) > 0))
       shell_cmd = shell;
    else
-      shell_cmd = strdup("/bin/sh");
+      shell_cmd = strdup("/bin/bash");
 
    if (session_name)
       snprintf(buf, sizeof(buf), "%s %s", session_cmd, session_name);
@@ -871,7 +871,7 @@
       syslog(LOG_NOTICE,
              "Neither '%s' or '/bin/sh' are working login shells for user '%s'. Your session may not function properly. ",
              shell, user);
-      shell_cmd = strdup("/bin/sh");
+      shell_cmd = strdup("/bin/bash");
    
    res = execlp(shell_cmd, shell_cmd, "-c", buf, NULL);
 
